fix(RouterLink): Prevent Disabled RouterLink from routing to top of page

lordkiz opened the pull request:
The type of this PR is: **Bugfix**

<!-- Bugfix/Feature/Enhancement/Documentation -->

<!-- If applicable, write the Jira ticket number in square brackets e.g. `[GRO-434]`
     The Jira integration will turn it into a clickable link for you. -->

This PR solves []

### Description
- When a RouterLink's `to` prop is passed a `null`, onClick automatically jumps to top of page. This PR fixes that.
<!-- Implementation description -->


[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

olerichter00 reviewed (approved):
Thanks for the fix 🤔 

But wasn't a fix like this already merged in https://github.com/artsy/force/pull/10675?

olerichter00 commented:
Ah, this PR fixes the issue for the metadata and https://github.com/artsy/force/pull/10675 only fixed the issue for the grid item itself.

lordkiz commented:
> But wasn't a fix like this already merged in #10675?

- When we pass null to `to`, since null is not one of our routes the RouterLink uses RouteUnawareLink which is a styled `a` tag. But it passes an empty string to this `a` tag which just takes the user to the top of the page. I just changed it to undefined.
- Also changed the cursor for Metadata

damassi commented:
@lordkiz, @olerichter00 - can you hover over a link and take a screenshot of the browser status bar and paste it in here? 

Does it say `http://whateverlink/undefined`? If so, we should revert this and fix the problem at its origin, which is in the `MetaData` component. If the link is undefined, we shouldn't wrap the component with a link. Having urls with `undefined` in them isn't great for SEO and other things. 

(Also, whenever changes are made to to code in the `System` folder, be sure to tag @dzucconi or me or @araujobarret as a reviewer on the PR please 🙏)

damassi commented on src/System/Router/RouterLink.tsx:
```diff
@@ -42,7 +42,9 @@ export const RouterLink: React.FC<RouterLinkProps> = React.forwardRef(
 
-    return <RouterUnawareLink href={to ?? ""} {...deprecated} {...rest} />
+    return (
+      <RouterUnawareLink href={to ?? undefined} {...deprecated} {...rest} />
```
☝️

lordkiz commented:
> @lordkiz, @olerichter00 - can you hover over a link and take a screenshot of the browser status bar and paste it in here?
> Does it say `http://whateverlink/undefined?`

@damassi No, it doesn't.

https://user-images.githubusercontent.com/18648835/183135837-28f91c59-a325-46da-82c7-2b7306a19ef9.mov


> (Also, whenever changes are made to to code in the System folder, be sure to tag @dzucconi or @damassi or @araujobarret as a reviewer on the PR please 🙏)

Noted. Sure. Thanks 🙏

dzucconi commented:
Why would we ever want a link to null?

damassi commented:
Yah @lordkiz we should fix that. We can conditionally render the `RouterLink` component vs using `undefined` and keep the behavior as it was.

dzucconi commented on src/Components/Artwork/Metadata.tsx:
```diff
@@ -65,7 +63,7 @@ export const Metadata: React.FC<MetadataProps> = ({
 const DisabledLink = styled(RouterLink)`
-  cursor: not-allowed;
+  cursor: default;
 `
```
This method of disabling the link is _not good_. We shouldn't have any empty anchor tag hanging out like this — it's still keyboard focusable.

dzucconi replied:
You should likely completely swap out the tag here.

dzucconi replied:
This is a good example of "fixing" the symptom rather than the root cause.

lordkiz commented:
@damassi @dzucconi This PR has been reverted here: https://github.com/artsy/force/pull/10714 🙏
